packages/cysync-core/src/components/AppUpdateBar.tsx

```diff
@@ -0,0 +1,171 @@
+import { UpdateInfo } from '@cypherock/cysync-interfaces';
+import {
+  Check,
+  CloudDownload,
+  LangDisplay,
+  UpdateBar,
+  UpdateState,
+} from '@cypherock/cysync-ui';
+import React, { FC, ReactNode, useEffect, useState } from 'react';
+import { useTheme } from 'styled-components';
+
+import { autoUpdater } from '~/utils';
+import logger from '~/utils/logger';
+
+import { selectLanguage, useAppSelector } from '..';
+
+enum AppUpdateState {
+  Checking,
+  Confirmation,
+  Downloading,
+  Successful,
+  Failed,
+}
+
+enum InternalAppUpdateState {
+  Checking,
+  Downloading,
+  Installing,
+}
+
+type UpdateBarType = {
+  [key in AppUpdateState]?: {
+    icon: ReactNode;
+    text: string;
+    updateState: UpdateState;
+    buttonText?: string;
+    onButtonClick?: () => void;
+  };
+};
+
+export const AppUpdateBar: FC = () => {
+  const theme = useTheme();
+  const lang = useAppSelector(selectLanguage);
+  const [state, setState] = useState<AppUpdateState>(AppUpdateState.Checking);
+  const [updateInfo, setUpdateInfo] = useState<UpdateInfo | undefined>();
+  const [downloadProgress, setDownloadProgress] = useState(57);
+  const [internalState, setInternalState] = useState(
+    InternalAppUpdateState.Checking,
+  );
+
+  const onError = (error: any) => {
+    logger.error('App update error');
+    logger.error(error);
+    setState(AppUpdateState.Failed);
+  };
+
+  const downloadUpdate = async () => {
+    try {
+      setInternalState(InternalAppUpdateState.Downloading);
+      setDownloadProgress(0);
+      setState(AppUpdateState.Downloading);
+      await autoUpdater.downloadUpdate();
+    } catch (error) {
+      onError(error);
+    }
+  };
+
+  const installUpdate = async () => {
+    try {
+      setInternalState(InternalAppUpdateState.Installing);
+      await autoUpdater.installUpdate();
+      setState(AppUpdateState.Successful);
+    } catch (error) {
+      onError(error);
+    }
+  };
+
+  const onRetry = () => {
+    const retryFuncMap: Record<InternalAppUpdateState, () => Promise<void>> = {
+      [InternalAppUpdateState.Checking]: checkForUpdates,
+      [InternalAppUpdateState.Downloading]: downloadUpdate,
+      [InternalAppUpdateState.Installing]: installUpdate,
+    };
```

Comment:
Most of the things in this file is common with the app update on onboarding. Try keeping all the common logic in a hook and use it here.